QuizData/uploadQuiz.js

```diff
@@ -0,0 +1,144 @@
+const express = require("express")
+const router = express.Router()
+
+const {Quiz} = require("./quiz.model.js")
+const {User} = require("../Users/user.model.js")
+
+router.route("/")
+  router.param("quizId", async (req,res,next,quizId) => {
+    try{
+      const findQuiz = await Quiz.findById(quizId)
+      if(!findQuiz){
+        return res.status(400).json({success: false, message: "Couldn't find quiz! Please enter valid quizId"})
+      }
+
+      req.quiz = findQuiz;
+      next();
+    }catch(error){
+      console.log(error)
+      res.status(400).json({success: false, message: "Something went wrong"})
+    }
+  })
+  router.param("userId", async (req,res,next,userId) => {
+    try{
+      const findUser = await User.findById(userId)
+      if(!findUser){
+        return res.status(400).json({success: false, message: "Couldn't find user! Please enter valid userId"})
+      }
+
+      req.user = findUser;
+      next();
+    }catch(error){
+      console.log(error)
+      res.status(400).json({success: false, message: "Something went wrong"})
+    }
+  })
+
+router.route("/")
+  .get(async (req,res) => {
+    try{
+      const quizzes = await Quiz.find({})
+      res.status(200).json({success: true, quizzes})
+    }catch(error){
+      res.status(400).json({success: false, error})
+    }
+  })
+  .post(async(req,res) => {
+    const {title,img} = req.body;
+
+    const quizzes = await Quiz.find({})
+
+    try{
+      const newQuiz = new Quiz({
+        title: title,
+        quizNo: quizzes.length + 1,
+        img: img,
+        highestScore: 0,
+        highScorerName: "",
+        questions: []
+      })
+
+      const saveQuiz = await newQuiz.save()
+      return res.status(201).json({success: true, user: saveQuiz})
+    }catch(error){
+      res.status(500).json({success: false, message: "Sorry! couldn't add quiz. Retry...", error})
+    }
+  })
+
+  router.route("/:quizId")
+    .post(async (req,res) => {
+      const {question,points,negativePoints,value1,isCorrect1, value2,isCorrect2, value3, isCorrect3, value4, isCorrect4} = req.body;
+
+      let {quiz} = req;
```

Comment:
Since you've already found the value in middleware, this could be const

Reply:
But I have to change its value below it.